aiida/restapi/translator/data/structure.py

```diff
@@ -95,7 +95,7 @@ def get_visualization_data(node, format=None, supercell_factors=[1, 1, 1]):
             except LicensingException as e:
                 response = e.message
 
-        else:
+        elif format == "chemdoodle":
             import numpy as np
```

Comment:
I know this was there before, but can we move this bit of logic and deal with it as with the other formats, instead of doing it specially here?
The way to do it is to add a `_prepare_chemdoodle` method to the StructureData class (aiida/orm/data/structure.py), see e.g. `_prepare_xyz`. You can move all the code generation in there.
Then chemdoodle will automatically be found among the formats, and you don't need a special case here.

Reply:
just one question: the `_prepare_chemdoodle` method should return a string (as all others _prepare_ methods) or I can still return the json dictionary?

Reply:
Mmm, good point. I think it should return a string.
Maybe you can then return a valid JSON string?

Reply:
Ok. I will add a JSON parse in the Javascript to retrieve the correct format to pass to chemdoodle.

Reply:
indeed, I didn't think to this.
I still think that the right thing to do is to return a string in AiiDA (as the method is called `_exportstring` and it might need to be stored on file).
To preserve back-compatibility with people using a REST API at a earlier version, can you put in the client code a test on the type: if it's already a JSON Object, just use it, while if it's a string, parse it first?

Reply:
Sure.